### PR TITLE
Stabilize Range[Inclusive]::is_empty

### LINKED ISSUE #48111: Tracking issue for Range[Inclusive]::is_empty (feature range_is_empty)
These methods are a clear way to check for empty ranges without the restrictions of `ExactSizeIterator`.

Merged as unstable on 2018-02-15 in PR at https://github.com/rust-lang/rust/pull/48087

### PR DESCRIPTION
**Pull request by scottmcm**
I would like to propose these two simple methods for stabilization:
- Knowing that a range is exhausted isn't otherwise trivial
- Clippy would like to suggest them, but had to do extra work to disable that path <https://github.com/rust-lang/rust-clippy/issues/3807> because they're unstable
- These work on `PartialOrd`, consistently with the stable `contains` method, and are thus more general than iterator-based approaches that need `Step`
- They've been unchanged for some time, and have picked up uses in the compiler
- Stabilizing them doesn't block any future iterator-based `is_empty` plans, as these inherent ones are preferred in name resolution

https://doc.rust-lang.org/nightly/std/ops/struct.Range.html#method.is_empty
https://doc.rust-lang.org/nightly/std/ops/struct.RangeInclusive.html#method.is_empty

Closes #48111

### CODE REVIEW

**Comment by scottmcm** [comment]
Since this needs an FCP, picking someone from libs who recently re-tagged the tracking issue:

r? @KodrAus

**Comment by KodrAus** [comment]
@rfcbot fcp merge

This proposes stabilizing the following API:

```rust
impl<Idx: PartialOrd<Idx>> Range<Idx> {
    pub fn is_empty(&self) -> bool;
}

impl<Idx: PartialOrd<Idx>> RangeInclusive<Idx> {
    pub fn is_empty(&self) -> bool;
}
```

**Comment by scottmcm** [comment]
@SimonSapin @sfackler @withoutboats Friendly ping for this pFCP.

**Comment by scottmcm** [comment]
Darn, missed 1.47 by a day, I guess.  I've changed the stability attributes to 1.48.

**Comment by Mark-Simulacrum** [comment]
Heh well we have not yet actually branched beta, that usually happens Tuesdays :)

But likely this wouldn't have landed in time anyway...

**Comment by scottmcm** [comment]
Doh, I mixed up `-` and `_`.  That'll teach me to submit and trust the PR instead of testing locally 😬

**Comment by scottmcm** [comment]
@bors r=dtolnay

Clippy is passing locally, so 🤞
image

File: library/core/src/ops/range.rs [hunk 1]
```diff
@@ -135,13 +133,11 @@ impl<Idx: PartialOrd<Idx>> Range<Idx> {
...
     ///
     /// ```
-    /// #![feature(range_is_empty)]
-    ///
     /// assert!(!(3.0..5.0).is_empty());
     /// assert!( (3.0..f32::NAN).is_empty());
     /// assert!( (f32::NAN..5.0).is_empty());
     /// ```
-    #[unstable(feature = "range_is_empty", reason = "recently added", issue = "48111")]
+    #[stable(feature = "range_is_empty", since = "1.47.0")]
```
Comment (lzutao) [inline library/core/src/ops/range.rs:140]:
```suggestion
    #[stable(feature = "range_is_empty", since = "1.48.0")]
```

Comment (Mark-Simulacrum) [reply library/core/src/ops/range.rs:140]: It is still on track for 1.47, though barely.

Comment (scottmcm) [reply library/core/src/ops/range.rs:140]: Merge to beta hasn't happened yet: https://github.com/rust-lang/rust/commits/beta

Will absolutely do that if the beta goes in before this one, @lzutao

Comment (lzutao) [reply library/core/src/ops/range.rs:140]: the milestone was changed to 1.48.

**Comment by scottmcm** [comment]
Wow am I having good luck today:
image

@bors retry

File: library/core/src/ops/range.rs (hunk 1 above)
Comment (scottmcm) [reply library/core/src/ops/range.rs:140]: I set it back to 1.47.

**Pull request merged**